### PR TITLE
Fix bug in wave interpolated post config file 

### LINKED ISSUE #1658: Wave post fails for mx025 (and possibly others)
**Expected behavior**
Wave post should run for all supported wave grids.

**Current behavior**
Wave post fails for mx025 (and possibly other grids).

**Machines affected**
All

**To Reproduce**
1. Set up a C384 coupled experiment or manually change the `waveGRD` to 'mx025'
```
./setup_expt.py gfs forecast-only --app S2SWA --resdet 384 --comrot "${NOSCRUB}/global-workflow" --expdir "${SAVE}/global-workflow/expdir" --idate 2013040100 --edate 2013040100 --pslot "${pslot}"
```
2. Observe wave post failing

**Context**
This surfaced after the default wave grid was updated to `mx025`. The problem may be briefly hidden after PR #1653 is merged as it changes the default wave grid based on the atm resolution and `mx025` is only used for C768+, which is rarely used in workflow development.

**Detailed Description**
Running develop out-of-the-box at a resolution that uses 'mx025' (and quite possibly others) fails due to `GRDNAME` never being defined:

```
/work2/noaa/global/wkolczyn/save/global-workflow/develop/scripts/exgfs_wave_post_gridded_sbs.sh: line 405: GRDNAME: unbound variable
```

There appears to be two things going on:
- `mx025` is not in either of the `case` blocks that may define `GRDNAME`.
- Both of those case blocks are contained within both an if and a for loop of `$DOGRI_WAV`/`$waveinterpGRD` or `$DOGRB_WAV`/`$wavepostGRD`, but is being used outside of those blocks.

**Additional Information**

**Possible Implementation**

### PR DESCRIPTION
**Pull request by JessicaMeixner-NOAA**
**Description**

Update logic for interpolated wave gridded post 

Fixes #1658

**Type of change**

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Two tests were completed, one with C48 S2SW forecast with the glo_025, ie no interpolated wave output: /scratch1/NCEPDEV/climate/Jessica.Meixner/gwissue1658/makefix/c48o500w500t2
and another test to show the test reported in the issue was fixed: /scratch1/NCEPDEV/climate/Jessica.Meixner/gwissue1658/makefix/test02

  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

### CODE REVIEW

File: parm/config/gfs/config.wavepostsbs [hunk 1]
```diff
@@ -17,7 +17,7 @@
 export DOFLD_WAV='YES' # Field data
 export DOPNT_WAV='YES' # Station data
 export DOGRB_WAV='YES' # Create grib2 files
-if [[ -z ${waveinterpGRD} ]]; then
+if [[ ! -z "${waveinterpGRD}" ]]; then
```
[the remark this answers is not in the document]
Comment (aerorahul) [reply parm/config/gfs/config.wavepostsbs]:
```suggestion
if [[ -n "${waveinterpGRD}" ]]; then
```

File: parm/config/gfs/config.wavepostsbs [hunk 2]
```diff
@@ -17,7 +17,7 @@ export DOIBP_WAV='NO' # Input boundary points
 export DOFLD_WAV='YES' # Field data
 export DOPNT_WAV='YES' # Station data
 export DOGRB_WAV='YES' # Create grib2 files
-if [[ -z ${waveinterpGRD} ]]; then
+if [[ -n "${waveinterpGRD}" ]]; then
```
Comment (aerorahul) [inline parm/config/gfs/config.wavepostsbs:20]: Where else is the variable `waveinterpGRD` used?

If we just do this here:
```
if [[ -n "${waveinterpGRD:-}" ]]; then
```

If it is only here, then there is no need to define empty strings in `config.wave`.
Line 89 in `config.wave` is also meaningless as the proper value for `waveinterpGRD` is being set in the `case` block above.

Comment (JessicaMeixner-NOAA) [reply parm/config/gfs/config.wavepostsbs:20]: waveinterpGRD is used in multiple scripts, for example a quick grep shows other places this is used: 

scripts/exgfs_wave_post_gridded_sbs.sh:72:  export waveinterpGRD=${waveinterpGRD?Var wavepostGRD Not Set}
scripts/exgfs_wave_post_gridded_sbs.sh:81:  echo "   Interpolated grids : $waveinterpGRD"
scripts/exgfs_wave_post_gridded_sbs.sh:105:  for grdID in ${waveGRD} ${wavepostGRD} ${waveinterpGRD}; do
scripts/exgfs_wave_post_gridded_sbs.sh:116:  for grdID in ${waveGRD} ${wavepostGRD} ${waveinterpGRD}; do
scripts/exgfs_wave_post_gridded_sbs.sh:140:    for intGRD in $waveinterpGRD
scripts/exgfs_wave_post_gridded_sbs.sh:169:    for grbGRD in $waveinterpGRD $wavepostGRD
scripts/exgfs_wave_post_gridded_sbs.sh:279:        for grdID in $waveinterpGRD
scripts/exgfs_wave_init.sh:82:  array=($WAVECUR_FID $WAVEICE_FID $WAVEWND_FID $waveuoutpGRD $waveGRD $waveesmfGRD $wavepostGRD $waveinterpGRD)

**Review by WalterKolczynski-NOAA — APPROVED** [review]
I believe you that it does, but could you please explain how correcting this setting prevents an uninitialized `GRDNAME` from being used at https://github.com/NOAA-EMC/global-workflow/blob/develop/scripts/exgfs_wave_post_gridded_sbs.sh#L409, since it doesn't appear to be within an if-block for `$DOGRI_WAV`.

**Comment by JessicaMeixner-NOAA** [comment]
> I believe you that it does, but could you please explain how correcting this setting prevents an uninitialized `GRDNAME` from being used at https://github.com/NOAA-EMC/global-workflow/blob/develop/scripts/exgfs_wave_post_gridded_sbs.sh#L409, since it doesn't appear to be within an if-block for `$DOGRI_WAV`.

Before, DOGRI_WAV was indvertedly and incorrectly sat to NO  when using waveGRD=mx025.  When DOGRI_WAV=YES, we go into the section of code here: https://github.com/NOAA-EMC/global-workflow/blob/develop/scripts/exgfs_wave_post_gridded_sbs.sh#L276-L305 where we then set the following values: 
 reg025) GRDNAME='global' ; GRDRES=0p25 ; GRIDNR=255  ; MODNR=11 ;;


More evidence can be found in the log file where it is running as expected here: /scratch1/NCEPDEV/climate/Jessica.Meixner/gwissue1658/makefix/test02/COMROOT/test02/logs/2013040100/gfswavepostsbs.log
and compare that to your log file or a previous log file from my first try where it was not working here: /scratch1/NCEPDEV/climate/Jessica.Meixner/gwissue1658/test01/COMROOT/test01/logs/2013040100/gfswavepostsbs.log.3

File: parm/config/gfs/config.wavepostsbs (hunk 2 above)
Comment (aerorahul) [reply parm/config/gfs/config.wavepostsbs:20]: Thanks @JessicaMeixner-NOAA 
Its (`waveinterpGRD`) use in `scripts/exgfs_wave_post_gridded_sbs.sh` is handled by the change in `config.wavepostsbs` as that config file is sourced in the job JGLOBAL_WAVE_POST_SBS.

Its use in `scripts/exgfs_waveinit.sh` should be handled as:
```
 array=($WAVECUR_FID $WAVEICE_FID $WAVEWND_FID $waveuoutpGRD $waveGRD $waveesmfGRD $wavepostGRD ${waveinterpGRD:-})
```
Infact, looking at this, likely all(?) these variables should have a default `:-` and the lines after the case block in `config.wave` are no longer relevant or necessary.

Comment (JessicaMeixner-NOAA) [reply parm/config/gfs/config.wavepostsbs:20]: Noted, perhaps we can open a separate issue to address these concerns as I feel like these are out of the scope of this PR which is to simply fix the bug reported in issue #1658 which I believe I have proved the update here does through answering questions and the tests.

Comment (aerorahul) [reply parm/config/gfs/config.wavepostsbs:20]: Feel free to open an issue and assign yourself if you would be comfortable improving the state of the wave configuration and use in the workflow.  If not, leave it open as a reminder of the issue and we will address it as resources permits.

**Comment by WalterKolczynski-NOAA** [comment]
> > I believe you that it does, but could you please explain how correcting this setting prevents an uninitialized `GRDNAME` from being used at https://github.com/NOAA-EMC/global-workflow/blob/develop/scripts/exgfs_wave_post_gridded_sbs.sh#L409, since it doesn't appear to be within an if-block for `$DOGRI_WAV`.
> 
> Before, DOGRI_WAV was indvertedly and incorrectly sat to NO when using waveGRD=mx025. When DOGRI_WAV=YES, we go into the section of code here: https://github.com/NOAA-EMC/global-workflow/blob/develop/scripts/exgfs_wave_post_gridded_sbs.sh#L276-L305 where we then set the following values: reg025) GRDNAME='global' ; GRDRES=0p25 ; GRIDNR=255 ; MODNR=11 ;;
> 
> More evidence can be found in the log file where it is running as expected here: /scratch1/NCEPDEV/climate/Jessica.Meixner/gwissue1658/makefix/test02/COMROOT/test02/logs/2013040100/gfswavepostsbs.log and compare that to your log file or a previous log file from my first try where it was not working here: /scratch1/NCEPDEV/climate/Jessica.Meixner/gwissue1658/test01/COMROOT/test01/logs/2013040100/gfswavepostsbs.log.3

Okay. Is there no situation where both `$DOGRI_WAV` and `$DOGRB_WAV` are "NO"? The script currently allows it.

In general, I think we need some refactoring of wave post (not now though).

**Comment by JessicaMeixner-NOAA** [comment]
You should not be running this job if both DOGRI_WAV and DOGRB_WAV are NO.

**Review by aerorahul — APPROVED** [review]

**Pull request merged**